src/viam/sdk/module/signal_manager_posix.cpp

```diff
@@ -0,0 +1,29 @@
+#include <viam/sdk/module/signal_manager.hpp>
+
+#include <csignal>
+#include <pthread.h>
+#include <signal.h>
+
+namespace viam {
+namespace sdk {
+
+struct SignalManager::Impl {
+    sigset_t sigset_;
+};
+
+SignalManager::SignalManager() : pImpl(std::make_unique<Impl>()) {
+    sigemptyset(&(pImpl->sigset_));
+    sigaddset(&(pImpl->sigset_), SIGINT);
+    sigaddset(&(pImpl->sigset_), SIGTERM);
+    pthread_sigmask(SIG_BLOCK, &(pImpl->sigset_), NULL);
+}
+
+SignalManager::~SignalManager() {}
```

Comment:
```suggestion
SignalManager::~SignalManager = default;
```
try looking up this syntax if you're not familiar with it!